python/src/trezorlib/transport/thp/protocol_v2.py

```diff
@@ -133,13 +135,18 @@ def _send_channel_allocation_request(self, nonce: bytes):
         )
 
     def _read_channel_allocation_response(
-        self,
-        expected_nonce: bytes,
+        self, expected_nonce: bytes, retry: bool = False
     ) -> tuple[int, bytes]:
-        header, payload = self._read_until_valid_crc_check()
-        if not self._is_valid_channel_allocation_response(
-            header, payload, expected_nonce
-        ):
+        while True:
```

Comment:
Defensive programming: want to limit the number of retries? Say, to the approximate maximum number of (one-frame message) retransmits?